app/models/spree/order_decorator.rb

```diff
@@ -30,20 +30,14 @@
   checkout_flow do
     go_to_state :address
     go_to_state :delivery
-    go_to_state :payment, :if => lambda { |order|
-      # Fix for #2191
-      if order.shipping_method.andand.delivery?
-        if order.ship_address.andand.valid?
-          order.create_shipment!
-          order.update_totals
-        end
-      end
+    go_to_state :payment, if: ->(order) {
+      order.update_totals
       order.payment_required?
     }
     # NOTE: :confirm step was removed because we were not actually using it
     # go_to_state :confirm, :if => lambda { |order| order.confirmation_required? }
     go_to_state :complete
-    remove_transition :from => :delivery, :to => :confirm
+    remove_transition from: :delivery, to: :confirm
```

Comment:
if that is the only thing we modify, isn't there a way from Spree to just remove this transition to avoid having to duplicate all the other steps? Having that single line will make it immediately obvious that that is the only change.

Reply:
Good point. I had another read about the Spree state machine and added another commit.